lib/mix/lib/mix/tasks/test.ex

```diff
@@ -327,6 +327,13 @@ defmodule Mix.Tasks.Test do
   def run(args) do
     {opts, files} = OptionParser.parse!(args, strict: @switches)
 
+    files =
+      if Mix.Task.recursing?() do
+        Enum.map(files, &String.trim_leading(&1, "apps/#{Mix.Project.config()[:app]}/"))
```

Comment:
Looks great! We also need to remove any file that starts with "apps/" but does not belong to the current app. :)

Reply:
Since those won't match any files, should we leave them so they will appear in this line:
https://github.com/elixir-lang/elixir/blob/master/lib/mix/lib/mix/tasks/test.ex#L427
in the case that nothing matches?